snuba/datasets/storages/processors/replaced_groups.py

```diff
@@ -71,12 +83,27 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
                     query.add_condition_to_ast(
                         not_in_condition(
                             FunctionCall(
-                                None, "assumeNotNull", (Column(None, None, "group_id"),)
+                                None,
+                                "assumeNotNull",
+                                (Column(None, None, "group_id"),),
                             ),
                             [Literal(None, p) for p in exclude_group_ids],
                         )
                     )
-            else:
-                set_final = final
 
         query.set_from_clause(replace(query.get_from_clause(), final=set_final))
+
+    def _query_overlaps_replacements(self, query: Query, project_ids: Set[int]) -> bool:
+        """
+        Given a Query and a set of project ids the query "touches", returns whether
+        or not this Query's time range overlaps with any replacement for any of the
+        projects.
+        """
+        query_from, _ = get_time_range(query, "timestamp")
```

Comment:
I don't think this logic is correct. Shouldn't we be comparing `query_to` instead of `query_from`? Then the check would be `return latest_replacement_time <= query_to` or "did the replacement happen before or after the time period the query is querying".

Reply:
From my understanding, a replacement could be touching data an unknown amount of time back. So if a replacement happens at a certain time, any query looking at data before that time would be "overlapping". For now, a query doesn't need final if it's looking at data that has no replacements yet.

`.......Time ->....`


Query doesn't overlap:

`...Rep..From..To..` <- This case would say the query does overlap if we `return latest <= query_to`


Query overlaps:

`..From..Rep..To...` 
`.From..To..Rep....` <- This case would say the query doesn't overlap if we `return latest <= query_to`

If my understanding is incorrect, I can change the logic.